scripts/check_socialapi_code_quality.sh

```diff
@@ -1,13 +1,22 @@
 #!/bin/bash
 
+set -o errexit
+
 #  make relative paths work.
 cd $(dirname $0)/..
 
-v=$(git diff-tree -r --name-only --no-commit-id master `git rev-parse HEAD` ./go/src/socialapi 2>&1)
-if [ -z "$v" ]; then
-    echo "nothing has changed under ./go/src/socialapi"
-    exit 0
-fi
+git diff-tree -r --exit-code --name-only --no-commit-id HEAD \
+    go/src/socialapi && exit 0
+
+export GOPATH=${GOPATH:-$(pwd)/go}
+
+echo "checking deadcode"
+ls --directory $GOPATH/src/socialapi/*/**/**/ | \
+  xargs go/bin/deadcode
+
+echo "checking unused variables"
+echo $(cd $GOPATH/src; ls --directory socialapi/*/**/**/) | \
+  xargs $GOPATH/bin/varcheck
```

Comment:
That's what came to my mind first too. I assumed there's a catch for third level subdirectories.